iceoryx_hoofs/test/moduletests/test_posix_timer.cpp

```diff
@@ -379,57 +378,44 @@ TEST_F(Timer_test, GetOverrunsFailsWithNoCallback)
 }
 
 TIMING_TEST_F(Timer_test, CatchUpPolicySkipToNextBeatContinuesWhenCallbackIsLongerThenTriggerTime, Repeat(5), [&] {
-    std::atomic_bool hasTerminated{false};
-    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
-        [&](const iox::Error, const std::function<void()>, const iox::ErrorLevel) { hasTerminated = true; });
-
     Timer sut(TIMEOUT, [] { std::this_thread::sleep_for(std::chrono::milliseconds(TIMEOUT.toMilliseconds() * 10)); });
-
     ASSERT_FALSE(sut.start(Timer::RunMode::PERIODIC, Timer::CatchUpPolicy::SKIP_TO_NEXT_BEAT).has_error());
-
     std::this_thread::sleep_for(std::chrono::milliseconds(TIMEOUT.toMilliseconds() * 10));
-    TIMING_TEST_EXPECT_FALSE(hasTerminated);
+    // EXPECT_NO_DEATH
+    EXPECT_TRUE(true);
```

Comment:
Please revert the changes (in whole file). `cxx::Expects` / `cxx::Ensures` are calling the error handler and terminate the program. When this test fails the whole test is terminated and the user is wondering what happened.

If possible, a unit test should never die! We always want a list of all the failing unit tests with the condition/code line which failed - this test setup makes this impossible since we do not know if the test failed or some segfault occurred.

Please do this by dropping the commit otherwise we clutter or history with changes which actually never happened.

Reply:
You're right, `DEATH_TESTS` should be avoided. However, this is an intermediate step as the `posix::Timer` needs to be removed before `v2.0` e.g. by finishing off #754 or re-implementing it from scratch.